tests/test_sshfsmount.cpp

```diff
@@ -241,3 +247,107 @@ TEST_F(SshfsMount, unblocks_when_sftpserver_exits)
     mount.join();
     EXPECT_TRUE(stopped_ok);
 }
+
+TEST_F(SshfsMount, throws_when_unable_to_change_dir)
+{
+    bool invoked{false};
+    auto request_exec = make_exec_that_fails_for({"cd"}, invoked);
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    EXPECT_THROW(make_sshfsmount(), std::runtime_error);
+    EXPECT_TRUE(invoked);
+}
+
+TEST_F(SshfsMount, throws_when_unable_to_get_current_dir)
+{
+    bool invoked{false};
+    auto request_exec = make_exec_that_fails_for({"pwd"}, invoked);
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    EXPECT_THROW(make_sshfsmount(), std::runtime_error);
+    EXPECT_TRUE(invoked);
+}
+
+TEST_F(SshfsMount, executes_commands)
+{
+    std::unordered_map<std::string, std::string> commands = {
+        {"which sshfs", "/usr/bin/sshfs"},
+        {"pwd", "/home/ubuntu"},
+        {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
+         "/home/ubuntu/"},
+        {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && mkdir -p \"target\"'", ""},
+        {"id -nu", "ubuntu"},
+        {"id -ng", "ubuntu"},
+        {"id -u", "1000"},
+        {"id -g", "1000"},
+        {"sudo sshfs -o slave -o nonempty -o transform_symlinks -o allow_other :\"source\" \"target\"", "don't care"}};
+
+    std::unordered_map<std::string, std::string> executed;
+
+    bool invoked{false};
+    std::string output{"1000"};
+    auto remaining = output.size();
+    auto channel_read = make_channel_read_return(output, remaining, invoked);
+    REPLACE(ssh_channel_read_timeout, channel_read);
+
+    auto request_exec = [&commands, &executed, &remaining, &output, &invoked](ssh_channel, const char* raw_cmd) {
+        std::string cmd{raw_cmd};
+        for (auto command : commands)
+        {
+            if (executed.find(command.first) == executed.end() && cmd.find(command.first) != std::string::npos)
+            {
+                output = command.second;
+                remaining = output.size();
+                executed.insert(command);
+                invoked = true;
+                break;
+            }
+        }
+
+        return SSH_OK;
+    };
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    make_sshfsmount();
```

Comment:
I'd be explicit about `target` here, otherwise I wasn't sure where "target" comes from in the commands matched above.

Reply:
I'm not sure of getting your point here. 'target' is a string representing the mount point on the commands above, but I don't understand how to be explicit: using a variable to represent the mount point?

Reply:
I just meant `make_sshfsmount("target")` here - or maybe even better, `make_sshfsmount("different/target")` :)

Reply:
OK, now I understand. It makes sense.

Reply:
Done